### PR TITLE
Fix import bug when upgrading from 5.0.3 to 5.1.0 preview-1

### PR DESCRIPTION
**Pull request by derwiath**
This solves a bug that appeared when upgrading the engine from `5.0.3` to `5.1.0-preview1`.

The bug had to to with files that was moved between directories where *only* case differed between the source and target.
`ueimporter` had some code to handle this in a "smart" way, by just renaming the parent folder in case there were several files that moved from a directory with the old cased to the new case. This code was a bit too smart for it's own good, and did not work in all cases. Instead of trying to fix this, I removed the code completely and instead relied on the default behavior of moving the files one by one.

Want more details? Read on.

`ueimporter` figures out what it needs to do with the help of `git`, in this case it was a `CEF` folder that caused a ruccus.
```sh
$ git diff --name-status 5.0.3-release 5.1.0-preview1 -- Engine/Source/ThirdParty/CEF3/Build
D	Engine/Source/ThirdParty/CEF3/Build/Linux/Toolchain-clang.cmake
D	Engine/Source/ThirdParty/CEF3/Build/Linux/build.sh
A	Engine/Source/ThirdParty/CEF3/Build/README.md
A	Engine/Source/ThirdParty/CEF3/Build/linux/.devcontainer/devcontainer.json
A	Engine/Source/ThirdParty/CEF3/Build/linux/Readme.md
A	Engine/Source/ThirdParty/CEF3/Build/linux/Toolchain-clang.cmake
A	Engine/Source/ThirdParty/CEF3/Build/linux/build.bat
A	Engine/Source/ThirdParty/CEF3/Build/linux/cef_build.sh
A	Engine/Source/ThirdParty/CEF3/Build/macOS/README.md
A	Engine/Source/ThirdParty/CEF3/Build/macOS/cef_build.sh
A	Engine/Source/ThirdParty/CEF3/Build/macOS/update_drop.sh
A	Engine/Source/ThirdParty/CEF3/Build/windows/Readme.md
A	Engine/Source/ThirdParty/CEF3/Build/windows/build.bat
A	Engine/Source/ThirdParty/CEF3/Build/windows/cef_build.bat
```

The first letter `D` that a file has been moved and `A` that a file that has been added. A delete + add of the same file where only the case of parent folders changed is replaced with a single move operation. Often `git` can detect when a file has been moved, and list it as a `R` for rename, but in this case it failed to do so.

Back to the problem. Here we see that `Linux/Toolchain-clang.cmake` moved to `linux/Toolchain-clang.cmake`, this triggered the faulty parent case rename code which moved the entire `Linux` folder to `linux` as the first step.
This included `build.sh` that also exists in `Linux` and was marked as `D` here. When the turn came to delete it the file didn't exist on disk under `Linux` anymore, and the import process failed.

After this change `ueimporter` will process the all changes in a defined order; adds, deletes, modifications and moves. So, `build.sh` will be removed from `Linux` subfolder before the `ToolChain-clan.cmake` is moved out of there.

This is fixed by the first four changesets. I choose to also remove the parent case rename code as well, as it was broken, it never considered if all the files that lived in the folder with the wrong case was supposed to move to the new location. In this case it could have been that only `ToolChain-clan.cmake` was supposed to move, and not `build.sh`.

### CODE REVIEW

File: ueimporter/job.py [hunk 1]
```diff
@@ -46,38 +26,79 @@ def create_jobs(changes, plastic_repo, source_root_path, pretend, logger):
...
+            assert job_changes != None
+            job_changes.append(change)
+
+    logger.deindent()
+
+    # Convert Move to Modify if move appears to already happened
+    # in the target repo
+    logger.log('Finding moves that have already been applied in target.')
+    logger.indent()
+    moves_already_existing_in_target = [ \
```
Comment (mmmeri) [inline ueimporter/job.py:61]: This backslash is superfluous, right?

File: ueimporter/main.py [hunk 2]
```diff
@@ -405,6 +413,24 @@ def main():
     if not config.pretend and not verify_plastic_repo_state(config, logger):
         return 1
 
+    if not path_util.is_directory_on_case_sensitive_filesystem(
+            config.plastic_repo.workspace_root):
+        logger.log_warning('Warning: Case insensitive filesystem detected.\n'
+                           'ueimporter have no way of correctly replicating'
```
Comment (mmmeri) [inline ueimporter/main.py:419]: ueimporter *has* no way

File: ueimporter/op.py [hunk 3]
```diff
@@ -102,42 +95,32 @@ def __init__(self, change):
...
+            # we might still have a valid move, if the target file
+            # already exists. in this case no plastic move will be performed,
+            # but the contents of the file will be copied from source.
+            return OpValidation.invalid_not_exist(
+                self.filename,
+                target_root)
+        if target_exist_in_target and source_exist_in_target:
+            # Even though the target file already exist in the target root,
+            # we might still have a valid move, if the source file
+            # does not exist. In this case no plastic move wi
```
Comment (mmmeri) [inline ueimporter/op.py:121]: comment is truncated

File: ueimporter/main.py [hunk 4]
```diff
@@ -405,6 +413,24 @@ def main():
...
+                           ' case changes of files and directories from git.\n'
+                           'You are adviced to run ueimporter from an OS'
```
Comment (mmmeri) [inline ueimporter/main.py:421]: advised

**Review by mmmeri — COMMENTED** [review]
Looking good, just minor things :)

File: ueimporter/job.py (hunk 1 above)
Comment (derwiath) [reply ueimporter/job.py:61]: I guess it is, the interpreter should be able to use the brackets to figure out that the line continues.

I will remove it

File: ueimporter/main.py (hunk 2 above)
Comment (derwiath) [reply ueimporter/main.py:419]: Thank you, my english is ever improving :)

**Review by mmmeri — APPROVED** [review]

**Pull request merged**